packages/mobile/src/localCurrency/convert.ts

```diff
@@ -48,5 +48,5 @@ export function convertLocalAmountToDollars(
 //   If the user enters `2.99` cUSD, this function has no impact
 //   and `2.99` is still displayed in the confirmation screen.
 export function convertDollarsToMaxSupportedPrecision(amount: BigNumber) {
-  return new BigNumber(amount.toPrecision(18, BigNumber.ROUND_UP))
+  return new BigNumber(amount.toFixed(18, BigNumber.ROUND_UP))
```

Comment:
This was all my bad, I misunderstood the behavior of `toPrecision` when I implemented this.

It doesn't limit the number to X decimals, but to X significant digits.
See https://mikemcl.github.io/bignumber.js/#toP and https://en.wikipedia.org/wiki/Significant_figures#Significant_figures_rules_explained

`toFixed` does what we want.